code/Serving_API.py

```diff
@@ -45,6 +48,22 @@ def get_all_standard_paths(db_file):
     conn.close()
     return datas
 
+# fetch all asn nodes related to amazon from database
+def get_all_asn_node(db_file):
+    """read asn node from database"""
+    conn = sqlite3.connect(db_file)
+    cursor = conn.cursor()
+    sql_query = """
+    SELECT DISTINCT al.latitude, al.longitude
+    FROM asn_asname aa
+    JOIN asn_loc al ON aa.asn = al.asn
+    WHERE aa.asn_name LIKE '%amazon%' and al.physical_presence = 'True';
```

Comment:
We need an argument into this function to tell apart "amazon" and "google cloud". I'll look into what's a good query string for google cloud, but it might make sense to call this function twice and store the as location lists from both amazon and Google. Also update the function comment to remove reference specific to amazon.

Reply:
This is also where you want to be careful with your function name and what it does. In this case, it's really a `get_all_amazon_asn_nodes`.